tests/conftest.py

```diff
@@ -45,7 +47,7 @@ def inner(method, response_dict, status_code, headers=None):
         def mock(*args, **kwargs):
             return MockResponse(response_dict, status_code, headers=headers)
 
-        monkeypatch.setattr(requests, method, mock)
+        monkeypatch.setattr(requests, "request", mock)
```

Comment:
I changed how we call `requests`, rather than using `get_attr` to retrieve the function by name, always wall `request.request` and pass in the method